### PR TITLE
adding a feature to customize the egress rule

### PR DESCRIPTION
**Pull request by haidargit**
## what
* Added 4 new variables `egress_source_port`, `egress_dest_port`, `egress_protocol`, and `allowed_egress_cidr_blocks` for the `"aws_security_group_rule" "egress"` resource.
* By default, the egress rule `0.0.0.0/0` will be created. If user is expected to restrict outbound traffic, they can specify the required values.

## why
* We propose this PR because there are scenarios where users may want to restrict outbound traffic from their DocDB instances. 
* if the DocDB cluster is only used internally and do not need to communicate with wide systems or network, users may want to customize the egress rule for `0.0.0.0/0`. By providing the option to customize the egress rule, we are giving users a control over their security posture (compliance). For example, our docdb may only connected with internal applications inside the aws eks cluster, or users may integrate their cloud resources with a third party, such as _Prisma Cloud_ or maybe use _tfsec_ as their security scanner, which prompts users to kindly avoid `0.0.0.0/0` for security best practices.

## references 
* This PR will help to cover the minimum egress exposure of the security group, including: #34 

Thank you

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -9,11 +9,11 @@ resource "aws_security_group" "default" {
...
   count             = module.this.enabled ? 1 : 0
   type              = "egress"
-  description       = "Allow all egress traffic"
-  from_port         = 0
-  to_port           = 0
-  protocol          = "-1"
-  cidr_blocks       = ["0.0.0.0/0"]
+  description       = "Allow outbound traffic from CIDR blocks"
+  from_port         = var.egress_port
+  to_port           = var.egress_port
```
Comment (milldr) [inline main.tf:14]: please make these separate variables

Comment (haidargit) [reply main.tf]: new variables `egress_source_port` & `egress_dest_port` are added.
kindly check, thanks @milldr

Comment (haidargit) [reply main.tf]: @milldr

Comment (SebastianMCarreira) [reply main.tf]: from_port and to_port specify the range of ports allowed, not the destination and source ports.

Comment (haidargit) [reply main.tf]: Hi @SebastianMCarreira , can you help to elaborate more?
The terraform resource _["aws_security_group_rule"](https://registry.terraform.io/providers/hashicorp/aws/3.42.0/docs/resources/security_group_rule)_ has inputs _from_port_  and  _to_port_ , which behave like range of ports afaik. 

from this PR we propose to adjust the docdb egress port range to be dynamic, so the docdb can handle other than `from_port=0`,`to_port=0`, and `protocol="-1"` (too wide),

For example, a user can do this.
```
  from_port      = var.egress_source_port -> can be filled with 0
  to_port        = var.egress_dest_port --> can be filled with 65535
```

Are these new variable names needed to be changed?
`egress_source_port` & `egress_dest_port`

Thank you for your concern

Comment (haidargit) [reply main.tf]: @milldr 
PR updated. Modified new variables naming. Kindly review, thanks.

_make init_ ✔️ 
_make readme_ ✔️

Comment (milldr) [reply main.tf]: @haidargit looks like there were changes after `make readme` - ([failed test](https://github.com/cloudposse/terraform-aws-documentdb-cluster/actions/runs/5557065012/jobs/10227284934?pr=54))

please run one more time

Comment (haidargit) [reply main.tf]: Thanks @milldr 

`make init` & `make readme` have been re-run
PR updated, kindly check again

**Review by milldr — APPROVED** [review]
LGTM! Thanks for the contribution

**Comment by haidargit** [comment]
@milldr thanks Pal!

**Pull request merged**